drivers/adc/ad7616/ad7616.c

```diff
@@ -416,13 +647,100 @@ int32_t ad7616_read_data_serial(struct ad7616_dev *dev,
 	if (ret != 0)
 		return ret;
 
-	no_os_axi_io_write(dev->core_baseaddr, AD7616_REG_UP_CTRL, AD7616_CTRL_RESETN);
+	no_os_axi_io_write(dev->core_baseaddr, AD7616_REG_UP_CTRL,
+			   AD7616_CTRL_RESETN);
 
 	if (dev->dcache_invalidate_range)
 		dev->dcache_invalidate_range(msg.rx_addr, samples * 2);
 
 	return ret;
 }
+#endif
+
+/**
+ * @brief Read conversion result from device, checking with CRC if enabled.
+ * @param dev - ad7616_dev device handler.
+ * @param results - results buffer.
+ * @param samples - sample number.
+ * @return 0 in case of success, -1 otherwise.
+ */
+int32_t ad7616_read_data_serial(struct ad7616_dev *dev,
+				struct ad7616_conversion_result *results,
+				uint32_t samples)
+{
+	int32_t ret;
+
+	memset(results, 0, sizeof(*results) * samples);
+
+#ifdef PLATFORM_ZYNQMP
+	return ad7616_read_data_serial_zynqmp(dev, (void *)results, samples)
+#endif
+
+	for (uint32_t i = 0; i < samples; i++) {
+		ret = ad7616_toggle_conv(dev);
+		if (ret != 0)
+			return ret;
+
+		uint8_t crc;
+		ret = ad7616_read_channels(dev, results + (i * dev->layers_nb),
+					   &crc);
+		if (ret)
+			return ret;
+
+		if (dev->crc) {
+			ret = check_crc(results + (i * dev->layers_nb),
+					dev->layers_nb, crc);
+			if (ret) {
+				return ret;
+			}
+		}
+	}
+	return 0;
+}
+
+/**
+ * PAR read from device.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad7616_par_read(struct ad7616_dev *dev, uint8_t reg_addr,
+			uint16_t *reg_data)
+{
+#ifdef PLATFORM_ZYNQMP
+	uint32_t read;
+
+	no_os_axi_io_write(dev->core_baseaddr, AD7616_REG_UP_WRITE_DATA,
+			   0x0000 | ((reg_addr & 0x3F) << 9));
+	no_os_udelay(50);
+	no_os_axi_io_read(dev->core_baseaddr, AD7616_REG_UP_READ_DATA, &read);
+	*reg_data = read & 0xFF;
+	no_os_mdelay(1);
```

Comment:
hmmm, there is no return in this case. is the expected behavior to return -ENOSYS also for this platform or is there a missing return 0? see also in par_write API

Reply:
Well spotted that, that's a mistake